aws/resource_aws_kinesis_stream.go

```diff
@@ -85,14 +85,15 @@ func resourceAwsKinesisStreamCreate(d *schema.ResourceData, meta interface{}) er
 		StreamName: aws.String(sn),
 	}
 
-	_, err := conn.CreateStream(createOpts)
-	if err != nil {
-		if awsErr, ok := err.(awserr.Error); ok {
-			return fmt.Errorf("[WARN] Error creating Kinesis Stream: \"%s\", code: \"%s\"", awsErr.Message(), awsErr.Code())
+	err := resource.Retry(30*time.Second, func() *resource.RetryError {
+		_, err := conn.CreateStream(createOpts)
+		if isAWSErr(err, "LimitExceededException", "simultaneously be in CREATING or DELETING") {
+			return resource.RetryableError(err)
 		}
-		return err
-	}
+		return resource.NonRetryableError(err)
+	})
```

Comment:
Actually do you mind adding an error check here?

Reply:
Then I'm happy to click the merge button 😃

Reply:
@radeksimko done :)